src/main/scala/util/Cutpoint.scala

```diff
@@ -0,0 +1,93 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import chisel3._
+import firrtl.annotations.{Annotation, MultiTargetAnnotation, ReferenceTarget, Target}
+import firrtl.{AnnotationSeq, CircuitState, DependencyAPIMigration, LowForm, Transform}
+import chisel3.experimental.{ChiselAnnotation, annotate}
+import firrtl.annotations.TargetToken.{Instance, Ref}
+import firrtl.stage.RunFirrtlTransformAnnotation
+import firrtl.options.CustomFileEmission
+
+/** Marks a signal with a [[CutpointAnnotation]]. Optionally, a label can be
+  * passed in.
+  *
+  * The Verilog paths to each cutpoint will be emitted in a metadata file,
+  * called cutpoints.<label>.txt.
```

Comment:
should there also be an output file which is a list of all the cutpoints.<label>.txt files...?

Reply:
do you think that's something that should be added in right now?

I'm open to adding it in later if it's needed, but I think it's unnecessary for our current purposes.